### ISSUE TITLE
[FEATURE] Remap a jumper pin to ACT LED?

### ISSUE THREAD

**Issue by vintagepc** [keirf/greaseweazle-firmware#1]
Small feature request - can any of the pins in the jumper/SWD block be remapped as the "ACT" led pin?

I'd like to mount my greaseweazle in an external drive enclosure and I'd like to connect up the enclosure front LED to see the internal LED status.

**Comment by keirf** [comment]
It could I think be duplicated on TX, to which you could connect an LED and a pull-up resistor to 3v3?

**Comment by vintagepc** [comment]
The enclosure I have uses a common-ground two colour LED, sounds like that'd work just fine if I wire one colour to +3.3v for power and the other to the ACT remap with a pullup.  

Thanks for the quick response!

**Comment by keirf** [comment]
Okay here's a build you can try which will duplicate the LED on TX on Greaseweazle V4 boards. 
You need to be logged in to GitHub to see the Artifact download link at the bottom of the page. You can then take the HEX or the UPD file from the non-debug zip contained within. If you use the UPD file you can:
```
gw update <path/to/upd_file>
gw update --bootloader <path/to/upd_file>
```
https://github.com/keirf/greaseweazle-firmware/actions/runs/1693677912

**Comment by vintagepc** [comment]
Thanks! I'll give it a spin  later today, I hope!

**Comment by keirf** [comment]
Correct

**Comment by vintagepc** [comment]
Perfect, seems to be working as expected! I have a little additional transistor circuitry to wire up on my end for the dual colour stuff since the LED is common ground but just connecting up one of the colours it blinks in sync with ACT.

**Comment by jamescadd** [comment]
@vintagepc could you share what the dual color wiring looks like?  I'd like to hook up something similar.  Is this the case? https://www.thingiverse.com/thing:5031605

**Comment by vintagepc** [comment]
No, not that enclosure - I repurposed two old USB CDRW enclosures to build a custom one with an AC plug since I am using a dual 5.25/3.5 combo drive.

the LED is a standard common-ground red/green LED that came with the enclosure; I actually ended up giving up on the transistor circuitry when I realized it's far easier to just tweak the code and invert the GPIO output on TX so it is high when the LED should be on.  I used some series resistors to limit the current to ~5mA which is plenty bright. 

If you're sourcing a new LED, I'd suggest a common cathode version instead, that can be used as-is with this tweak. 

In my case, one side of the LED (green) is always wired to 3.3v /GND and therefore on, the red side is connected to TX so the LED effectively turns orange when the board is busy. If you don't like that effect one can use a transistor to shut off the green side when the red side is active to get a true red-green flip
